spec/services/ucas_matching/upload_matching_data_to_ucas_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe UCASMatching::UploadMatchingData, sidekiq: true do
+  describe '#perform' do
+    it 'uploads the latest file to UCAS' do
+      stub_request(:post, 'https://transfer.ucasenvironments.com/api/v1/token')
+        .to_return(
+          body: { access_token: '123456789' }.to_json,
+        )
+
+      request = stub_request(:post, 'https://transfer.ucasenvironments.com/api/v1/folders/685520099/files')
+
+      create(:submitted_application_choice)
+
+      UCASMatching::UploadMatchingData.perform_async
+
+      expect(request).to have_been_made
```

Comment:
Should we also check the authorization header here to make sure it's all hanging together?